MonoSingleton.cs

```diff
@@ -0,0 +1,26 @@
+using UnityEngine;
+
+namespace SubsurfaceStudios.Utilities.Instancing
+{
+    public abstract class MonoSingleton<T> : MonoBehaviour where T : MonoSingleton<T>
+    {
+        public static T Instance { get; private set; }
+
+        protected virtual void Awake()
+        {
+            if (Instance != null && Instance != this)
+            {
+                Destroy(gameObject);
```

Comment:
```suggestion
                Destroy(this);
```

You might want to switch this to use `this` instead, so that if we have multiple of these on one object it doesn't destroy the entire object.